practice/timezone/timezone/logic.py

```diff
@@ -0,0 +1,41 @@
+"""timezone Logic."""
+
+from tzwhere import tzwhere
+import arrow
+
+
+def get_time_at_tz(tz_string):
+    """Return the current time (as a string) at a given timezone."""
+    utc = arrow.utcnow()
+    time_string = str(utc.to(tz_string))
+    return time_string
+
+
+def get_tz(lat_lng_list):
+    r"""Return the time zone given lat, long.
+
+    >>> get_tz([45,45])
+    'Europe/Moscow'
+    """
+    tz = tzwhere.tzwhere()
+    tz_string = tz.tzNameAt(float(lat_lng_list[0]), float(lat_lng_list[1]))
+    return tz_string
+
+
+def get_time_at_latlng(tz_string):
+    """Return the current time at a given lat, long."""
+    return get_time_at_tz(tz_string)
+
+
+def get_requested_time(time_string):
+    """Return an arrow object with a given time and time zone."""
+    return arrow.get(time_string)
```

Comment:
This function deals with serialization. It converts the ISO string to a Arrow instance. Serialization functions should live in views, since they deal with what kind of interface the whole service presents. Logic is just verbs dealing with the native types of this problem.